cfpb/cfgov-refresh
# First pass at migrating Ask CFPB content

This focuses on simplifying knowledgebase data models and migrating
the content into Wagtail. The parent question-categories that we're using
as portal-page topics are broken out as categories, each having a set
of subcategories, for organizing answers and pages.
The core content bits -- questions and answers in two languages --
are consolidated into Answer objects that carry the English and Spanish
forms of a question, its answer, its slug, and metadata.

## Workflow
- An Answer object is where content editing is done. 
- Wagtail pages are created or updated based on workflow checkboxes for each language. If a box is checked, the relevant page will be updated with a draft status.
- Each language also has its own manual `last edited` field, intended to give us a more useful timestamp to track  editorial updates. Editors choose when an to mark an answer as updated, so minor metadata changes (or, say, a migration) won't pollute these values.
- Pages then follow our standard Wagtail review and approval workflow.
- The original Ask IDs get migrated to `ask_cfpb` as Answer IDs, which can be handy for searching both answers and pages. The answer page titles carry the ID as a prefix with a colon, so searching the Wagtail admin for "1979:" will find just the English and Spanish pages associated with that answer. 

## Additions

- ask_cfpb app

## Removals

- None. Knowledgebase remains a satellite app for now.

## Changes

- `settings/base.py` changed to add ask_cfpb to INSTALLED_APP

## Testing

These steps will get the current data migrated into the new models in Wagtail.
- create new db, load with recent prod data, run `migrate` and fire up runserver
```
./drop-db.sh
./create-mysql-db.sh
mysql -uroot v1 < ~/Downloads/production_django.sql
python cfgov/manage.py migrate
./runserver.sh
```

- Create a landing page under CFGov with title "Ask CFPB" and publish it to www
- If you want to view live and draft pages, edit your sites so default is localhost/port 8000
- Optionally add a sharing site for the default site
- in another terminal, open shell and migrate the knowledgebase data.

```
python cfgov/manage.py shell
```

```python
from ask_cfpb.migration_scripts.migrate_knowledgebase import migrate_knowledgebase
migrate_knowledgebase()
```

This take a few minutes to migrate the knowledgebase data into the new models and generate 2K+  English and Spanish pages.

You now should see the data tables in the Wagtail admin left sidebar under "Ask CFPB," and the pages should be live under `CFGov > Ask CFPB >` 

## Screenshots

<img width="358" alt="ask_in_wagtail" src="https://cloud.githubusercontent.com/assets/515885/23640657/1ced058e-02bd-11e7-8b2e-69b7f94c0561.png">

Try editing an Answer, selecting workflow update box and saving to update a page.
